spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Conv2D.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.nn.SpatialConvolution
+import com.intel.analytics.bigdl.nn.abstractnn.{Activity, DataFormat}
+import com.intel.analytics.bigdl.tensor._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+class Conv2D[T: ClassTag](
+  strides: Array[Int],
+  padding: String,
+  format: DataFormat = DataFormat.NHWC
+)(implicit ev: TensorNumeric[T]) extends Operation[Table, Tensor[T], T] {
+
+  override def updateOutput(inputs: Table): Tensor[T] = {
+    val input: Tensor[T] = inputs[Tensor[T]](1)
+    val filter: Tensor[T] = inputs[Tensor[T]](2)
+
+    val conv: SpatialConvolution[T] = format match {
```

Comment:
Should we consider caching the conv object?

Reply:
The filer is a input, it can be changed, so the filter size can be changed. They will be different convolution objects.